packages/api/src/router/widgets/dns-hole.ts

```diff
@@ -29,4 +30,18 @@ export const dnsHoleRouter = createTRPCRouter({
       integrationId: ctx.integration.id,
     };
   }),
+
+  enable: publicProcedure
+    .unstable_concat(createOneIntegrationMiddleware("interact", "piHole"))
+    .mutation(async ({ ctx }) => {
+      await new PiHoleIntegration(ctx.integration).enableAsync();
+    }),
+
+  disable: publicProcedure
+    .input(controlsInputSchema)
+    .unstable_concat(createOneIntegrationMiddleware("interact", "piHole"))
+    .mutation(async ({ ctx, input }) => {
+      const { duration } = input;
```

Comment:
I would simply use ``disableAsync(input.duration)`` instead. The deconstruct doesn't add much value here

Reply:
fixed